include/swift/Driver/Job.h

```diff
@@ -230,9 +230,20 @@ class CommandOutput {
 class Job {
 public:
   enum class Condition {
+    // There was no information about the previous build (i.e., an input map),
+    // or the map marked this Job as dirty
+    // but the input didn't change.
+    // Be maximally conservative with dependencies.
     Always,
+    // The input changed, or this job was non-cascading or [it was not dirty but
+    // primary output was missing].
+    // In other words, this job was scheduled as non-cascading in the last build
+    // but didn't get to run.
     RunWithoutCascading,
+    // The best case: input didn't change, output exists.
+    // Only run if it depends on some other thing that changed.
     CheckDependencies,
+    // Pessimal: Run no matter what (but may or may not cascade).
```

Comment:
I don't think "pessimal" is appropriate. As mentioned before, this is usually equivalent to RunWithoutCascading; it just means we don't compare against a previous swiftdeps file.

Reply:
OK